### ISSUE TITLE
gprofile error

### ISSUE THREAD

**Issue by dimitras** [smangul1/rop#110]
I run gprofile for the aligned reads (bam file) with this command: 
`python /SOFTWARE/rop/gprofile.py --mouse ropin/Aligned.bam ropout/gprofile/Aligned.gprofile.csv`
and I get the following error.

`
Load gene annotations ...
Load /SOFTWARE/rop/source/gprofile/annotations/mouse/bedPrepared/UTR3_NCBIM37_prepared.bed
Load /SOFTWARE/rop/source/gprofile/annotations/mouse/bedPrepared/UTR5_NCBIM37_prepared.bed
Load /SOFTWARE/rop/source/gprofile/annotations/mouse/bedPrepared/CDS_NCBIM37_prepared.bed
Load /SOFTWARE/rop/source/gprofile/annotations/mouse/bedPrepared/geneCoordinates_NCBIM37.bed
Open bam file ropin/Aligned.bam
Process chr 1
Traceback (most recent call last):
  File "/SOFTWARE/rop/gprofile.py", line 419, in <module>
    for read in bamfile.fetch(chr):
  File "pysam/calignmentfile.pyx", line 897, in pysam.calignmentfile.AlignmentFile.fetch (pysam/calignmentfile.c:10997)
  File "pysam/calignmentfile.pyx", line 819, in pysam.calignmentfile.AlignmentFile.parse_region (pysam/calignmentfile.c:10455)
ValueError: invalid reference `1`
`

I found a relative issue posted on svtyper, and I used samblaster to add the mate tags. Then I reran gprofile with the mated bam file, but the same error persists.

Please advise, thank you.

**Comment by smangul1** [comment]
Hi Dimitra,

Let me have a look and i will have an answer for you shortly

Thanks,
Serghei

On Thu, Jan 26, 2017 at 12:38 PM, Dimitra Sarantopoulou
<notifications@github.com> wrote:
> I run gprofile for the aligned reads (bam file) with this command:
> python /SOFTWARE/rop/gprofile.py --mouse ropin/Aligned.bam
> ropout/gprofile/Aligned.gprofile.csv
> and I get the following error.
>
> Load gene annotations ... Load
> /SOFTWARE/rop/source/gprofile/annotations/mouse/bedPrepared/UTR3_NCBIM37_prepared.bed
> Load
> /SOFTWARE/rop/source/gprofile/annotations/mouse/bedPrepared/UTR5_NCBIM37_prepared.bed
> Load
> /SOFTWARE/rop/source/gprofile/annotations/mouse/bedPrepared/CDS_NCBIM37_prepared.bed
> Load
> /SOFTWARE/rop/source/gprofile/annotations/mouse/bedPrepared/geneCoordinates_NCBIM37.bed
> Open bam file ropin/Aligned.bam Process chr 1 Traceback (most recent call
> last): File "/SOFTWARE/rop/gprofile.py", line 419, in <module> for read in
> bamfile.fetch(chr): File "pysam/calignmentfile.pyx", line 897, in
> pysam.calignmentfile.AlignmentFile.fetch (pysam/calignmentfile.c:10997) File
> "pysam/calignmentfile.pyx", line 819, in
> pysam.calignmentfile.AlignmentFile.parse_region
> (pysam/calignmentfile.c:10455) ValueError: invalid reference1
>
> I found a relative issue posted on svtyper, and I used samblaster to add the
> mate tags. Then I reran gprofile with the mated bam file, but the same error
> persists.
>
> Please advise, thank you.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.



-- 
Sincerely,
Serghei Mangul, Ph.D
QCB Postdoctoral Fellow,
Institute for Quantitative and Computational Biosciences,
Department of Computer Science and Human Genetics
University of California, Los Angeles
https://sergheimangul.wordpress.com

**Comment by Aditi2112j** [comment]
We are running svtyper on a linux system and facing this error:
Any suggestions as we are newbie to SVTyper.

Traceback (most recent call last):
  File "svtyper.py", line 1808, in <module>
    sys.exit(main())
  File "svtyper.py", line 1803, in main
    args.max_reads)
  File "svtyper.py", line 1528, in sv_genotype
    read_batch, many = gather_reads(sample, chromA, posA, ciA, z, read_batch, max_reads)
  File "svtyper.py", line 1315, in gather_reads
    min(pos + ci[1] + fetch_flank, chrom_length))):
  File "pysam/libcalignmentfile.pyx", line 855, in pysam.libcalignmentfile.AlignmentFile.fetch (pysam/libcalignmentfile.c:11188)
  File "pysam/libcalignmentfile.pyx", line 777, in pysam.libcalignmentfile.AlignmentFile.parse_region (pysam/libcalignmentfile.c:10647)
ValueError: invalid reference `1

Thank you in advance for the help and suggestions

**Comment by smangul1** [comment]
Can you please explain why do you need svtyper for ROP?

**Comment by Aditi2112j** [comment]
Its not for ROP. We used SVTyper to genotype the output from LUMPY. In the process we faced these issues. Any suggestions over how we can resolve them. 

File "svtyper.py", line 1528, in sv_genotype
read_batch, many = gather_reads(sample, chromA, posA, ciA, z, read_batch, max_reads)
File "pysam/libcalignmentfile.pyx", line 855, in pysam.libcalignmentfile.AlignmentFile.fetch (pysam/libcalignmentfile.c:11188)
File "pysam/libcalignmentfile.pyx", line 777, in pysam.libcalignmentfile.AlignmentFile.parse_region (pysam/libcalignmentfile.c:10647)
ValueError: invalid reference `1
